### PR TITLE
Raft Protocol vs Consul Protocol Doc Clarification

### PR DESCRIPTION
**Pull request by moofish32**
I do not know if I have this correct. I am unclear if `protocol` version and `raft` version are the same. Based on the info in `Specific Version Details` and `Compatibility Section` it's a little confusing. I'll do the editing if somebody can give a little clarification.

### CODE REVIEW

**Comment by slackpad** [comment]
Hey @moofish32 thanks for taking this up! The Raft protocol version is a totally separate and new thing for 0.8, so that would be worth making clear here. The Raft protocol version only affects servers and enables certain new autopilot features, so we could probably link to https://www.consul.io/docs/upgrade-specific.html#raft-protocol-version-compatibility for more details. Thinking about your confusion we could have maybe rolled it into the protocol version, but it's so specific and only affects servers, so we made it its own thing. After 0.9 it'll default to the latest and shouldn't need to be updated again (hopefully).

Also in the diff here, 0.8 doesn't support back to protocol version 1, we shed that in version 0.7.

File: website/source/docs/upgrade-specific.html.markdown [hunk 1]
```diff
@@ -49,7 +49,7 @@ When upgrading to Consul 0.8.0 from a version lower than 0.7.0, users will need
 set the [`-raft-protocol`](/docs/agent/options.html#_raft_protocol) option to 1 in
 order to maintain backwards compatibility with the old servers during the upgrade.
 After the servers have been migrated to version 0.8.0, `-raft-protocol` can be moved
-up to 2 and the servers restarted to match the default.
+up to 3 and the servers restarted to match the default.
```
Comment (moofish32) [inline website/source/docs/upgrade-specific.html.markdown]: @slackpad - Once all servers are migrated to 0.8 should the servers be restarted with 3 or 2?  Is there something special in Raft v2? I separated this commit, so it's easily dropped.

**Comment by moofish32** [comment]
@slackpad - I'm not a frontend guru, but I made a change to condense the table table a bit. Happy to go back and add in one line per version. I had one more question on the Raft version migration section.

File: website/source/docs/upgrade-specific.html.markdown (hunk 1 above)
Comment (slackpad) [inline website/source/docs/upgrade-specific.html.markdown]: This is a special case for people running pre-0.7 Consul versions, so 2 was right here.

**Comment by slackpad** [comment]
@moofish32 thanks for cleaning this up!

**Pull request merged**